### PR TITLE
FINERACT-1390:Withdrawal Charges as per PaymentType

### PR DESCRIPTION
**Pull request by rrpawar96**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java [hunk 1]
```diff
@@ -1076,16 +1076,19 @@ private void payWithdrawalFee(final BigDecimal transactionAmount, final LocalDat
         for (SavingsAccountCharge charge : this.charges()) {
 
             if (charge.isWithdrawalFee() && charge.isActive()) {
-                if (charge.getFreeWithdrawalCount() == null) {
-                    charge.setFreeWithdrawalCount(0);
-                }
 
-                if (charge.isEnableFreeWithdrawal()) {
-                    resetFreeChargeDaysCount(charge, transactionAmount, transactionDate, user);
+                if (paymentDetail.getPaymentType().getPaymentName().equals(charge.getCharge().getPaymentType().getPaymentName())) {
```
Comment (rrpawar96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java]: The logic of this PR is to allow withdrawal transactions by checking whether the payment type assigned to the charge is equal to the payment Type of the withdrawal transaction.

Comment (rrpawar96) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java]: condition 1: if the charge has both condition of specific paymentType and discountType (FINERACT-1374)
condition 2 : charge is assigned to specific payment type but normal charge(no discount Type)
condition 3:  discount charge but no condition of paymentType.
condition 4 : normal withdraw charges

**Comment by francisguchie** [comment]
@rrpawar96  I have tested this functionality and it works for me. Tried to put it in "live" production

**Comment by awasum** [comment]
This looks good.. Is there a way you could add a unit test to test the new paymentName parameter?

**Review by awasum — APPROVED** [review]
LGTM

**Comment by rrpawar96** [comment]
> This looks good.. Is there a way you could add a unit test to test the new paymentName parameter?

Yes, I will write a unit test for it

**Comment by rrpawar96** [comment]
@awasum @ptuomola  I have added a testcase for paymentType

**Pull request merged**